File: karaoke/api/routes/library.py
from karaoke.services.library_service import LibraryService

_library = LibraryService()


async def upload_file(query):
    return await _library.upload_file(query)


async def run_scan(body):
    return await _library.run_scan(body)


async def preview_scan(root, duplicate_policy, validate):
    return await _library.preview_scan(root, duplicate_policy, validate)


async def get_list(q: str = '', page: int = 1):
    return await _library.get_list(q, page)


async def delete_song(song_id, delete_disk=False):
    return await _library.delete_song(song_id, delete_disk)

File: static/js/domains/library.js
window.KTV = window.KTV || {};

(function (KTV) {
    var base = function () { return KTV.config.apiBase + '/library'; };

    KTV.library = {
        upload: function (formData) {
            return KTV.http.upload(base() + '/upload', formData);
        },
        scanPreview: function (root, duplicate_policy, validate) {
            return KTV.http.get(base() + '/scan/preview', {
                root: root,
                duplicate_policy: duplicate_policy,
                validate: validate,
            });
        },
        scan: function (body) {
            return KTV.http.post(base() + '/scan', JSON.stringify(body), 'application/json');
        },
        list: function (page, q) {
            return KTV.http.get(base() + '/songs', { page: page, q: q || '' });
        },
        remove: function (songId, deleteDisk) {
            return KTV.http.delete(base() + '/songs/' + songId, { delete_disk: deleteDisk });
        },
    };
})(window.KTV);
